software/control/widgets.py

```diff
@@ -1358,16 +1363,30 @@ def stop_recording(self):
 
 
 class NavigationWidget(QFrame):
-    def __init__(self, navigationController, slidePositionController=None, main=None, widget_configuration = 'full', *args, **kwargs):
+    def __init__(self, stage: AbstractStage, slidePositionController=None, main=None, widget_configuration = 'full', *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.navigationController = navigationController
+        self.log = squid.logging.get_logger(self.__class__.__name__)
+        self.stage = stage
         self.slidePositionController = slidePositionController
         self.widget_configuration = widget_configuration
         self.slide_position = None
-        self.flag_click_to_move = navigationController.click_to_move
+        # TODO(imo): Fix below.  Used to use NavigationController click flag tracking
+        self.flag_click_to_move = False
         self.add_components()
         self.setFrameStyle(QFrame.Panel | QFrame.Raised)
 
+        self.position_update_timer = QTimer()
```

Comment:
This is one way to do position updates.  It is nice because we only need to pass in pieces of the future `Microscope` API, and otherwise the widgets can be independent of each other.

An alternative, if we don't like the Timer/polling approach, is to define a standard collection of `Signal`  that any Widget can assume exists and pass them as an __init__ argument.  Then each Widget can just connect itself to any of those signals if needed at __init__ time, but doesn't need to be passed the entire machinery behind other Widgets.